src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +66,243 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
+			drone_desired_position(2) = _home_position_z - _follow_target_height;
 	}
 
 	return drone_desired_position;
 }
 
-Vector3f FlightTaskAutoFollowTarget::calculate_target_position_filtered(Vector3f pos_ned_est, Vector3f vel_ned_est,
-		Vector3f acc_ned_est)
-{
-	// Reset the smoothness filter once the target position estimate is available
-	if (!PX4_ISFINITE(_target_position_filter.getState()(0)) || !PX4_ISFINITE(_target_position_filter.getState()(1))
-	    || !PX4_ISFINITE(_target_position_filter.getState()(2))) {
-		_target_position_filter.reset(pos_ned_est);
-	}
 
-	// Low-pass filter on target position.
-	_target_position_filter.setParameters(_deltatime, POSITION_FILTER_ALPHA);
+float FlightTaskAutoFollowTarget::calculate_gimbal_height(float target_height) {
+	float gimbal_height{0.0f};
+
+	switch (_param_nav_ft_alt_m.get()) {
+		case FOLLOW_ALTITUDE_MODE_TRACK_TERRAIN:
+			// Point the gimbal at the ground level in this tracking mode
+			gimbal_height = _dist_to_ground;
+			break;
+
+		case FOLLOW_ALTITUDE_MODE_TRACK_TARGET:
+			// Point the gimbal at the target's 3D coordinates
+			gimbal_height = -_position(2) - target_height;
+			break;
 
-	if (_param_nav_ft_delc.get() == 0) {
-		_target_position_filter.update(pos_ned_est);
+		case FOLLOW_ALTITUDE_MODE_CONSTANT:
 
-	} else {
-		// Use a predicted target's position to compensate the filter delay to some extent.
-		const Vector3f target_predicted_position = predict_future_pos_ned_est(POSITION_FILTER_ALPHA, pos_ned_est, vel_ned_est,
-				acc_ned_est);
-		_target_position_filter.update(target_predicted_position);
-	}
+		//FALLTHROUGH
 
-	return _target_position_filter.getState();
+		default:
+			gimbal_height = _home_position_z -_position(2); // Assume target is at home position's altitude
+			break;
+	}
 
+	return gimbal_height;
 }
 
+
 bool FlightTaskAutoFollowTarget::update()
 {
 	_follow_target_estimator_sub.update(&_follow_target_estimator);
-	follow_target_status_s follow_target_status{};
+
+	follow_target_status_s follow_target_status{}; // Debugging uORB message for follow target status
 
 	if (_follow_target_estimator.timestamp > 0 && _follow_target_estimator.valid) {
-		const Vector3f pos_ned_est{_follow_target_estimator.pos_est};
-		const Vector3f vel_ned_est{_follow_target_estimator.vel_est};
-		const Vector3f acc_ned_est{_follow_target_estimator.acc_est};
-		const Vector3f target_position_filtered = calculate_target_position_filtered(pos_ned_est, vel_ned_est, acc_ned_est);
-		const Vector2f offset_vector_filtered = calculate_offset_vector_filtered(vel_ned_est);
-		const Vector3f drone_desired_position = calculate_drone_desired_position(target_position_filtered,
-							offset_vector_filtered);
+		// Update second order target pose filter, with a valid data
+		update_target_pose_filter(_follow_target_estimator);
 
-		// Set position and velocity setpoints
-		float desired_velocity_ff_scale = 0.0f;  // Used to ramp up velocity feedforward, avoiding harsh jumps in the setpoints
```

Comment:
Is `desired_velocity_ff_scale`, i.e. blending in the velocity-feedforward no longer required?

Reply:
blending no longer needed - he just sets the feed-forward according to the sum of lever arm velocity and target velocity, then MC position controller handles prioritization when near or far from position setpoint.

Reply:
- target velocity is filtered, so no jumps there.
- lever arm based velocity could make a step whenever orbit angle is changed because we immediately apply max orbiting rate, though @junwoo091400 added a ramp down to zero when close to the setpoint, and the orbit angle command is then no longer changed, so no more orbit rate based lever arm velocity.

A final output filter would solve this :) -- but im not sure it will be too deleterious anyway.. would be just like you change the go-to position, the velocity setpoint will have a step input and gradually decrease to zero when the orbit angle reaches it's commanded value.

@junwoo091400 did you notice anything bad looking from this in your recent flights? we could address if the behavior was poor

Reply:
I haven't tested with the orbital velocity bug fix version, will test today :hand:. Since the previous tests only had the orbital velocity setpoint of 1 m/s magnitude, it isn't a valid test case I believe.

Reply:
orbit angle trajectory generator has solved this problem :) Both acc & vel limited!